database/migrations/2023_12_13_093832_create_appointments_table.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('appointments', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('patient_id')->constrained('users')->onDelete('cascade');
+            $table->foreignId('doctor_id')->constrained('users')->onDelete('cascade');
+            $table->foreignId('department_id')->constrained('departments');
+
+            $table->string('patient_first_name');
```

Comment:
please rename `patient_first_name` to `first_name`. This field should be nullable. 
same for `last_name`

Reply:
@kolanaveen Why we require these fields to be nullable? Don't we require patient name?

Reply:
@DivyaPatange  If we choose the `patient`, we can retrieve the patient's information. Sometimes, appointments are made with patients who are not on our list. Therefore, when selecting a patient in the appointment creation form, hide the fields for `first_name` and `last_name`.

Reply:
@kolanaveen I agree sometimes appointments are made with patients who are not in out list. At this time, we can create new patient by the name provided from first_name and last_name. Also, if patient record is presented and if they want to change their first name and last name at this time i think we require those fields.

Reply:
@DivyaPatange Great idea! Let's include a feature to add new patients. Also, when choosing a patient, automatically fill in their first and last name.